WcaOnRails/app/controllers/posts_controller.rb

```diff
@@ -5,7 +5,13 @@ class PostsController < ApplicationController
   before_action -> { redirect_to_root_unless_user(:can_create_posts?) }, except: [:index, :rss, :show]
 
   def index
-    @posts = Post.where(world_readable: true).order(sticky: :desc, created_at: :desc).includes(:author).page(params[:page])
+    tag = params[:tag]
+    if tag
+      @posts = Post.joins(:post_tags).where('post_tags.tag = ?', tag)
+    else
+      @posts = Post.joins("LEFT JOIN post_tags ON post_tags.post_id = posts.id").group("posts.id").having("IFNULL(SUM(tag = 'wdc'), 0) = 0")
```

Comment:
Ha! Yeah, this was impossible to write, so I'm not surprised it's impossible to read. The idea is to remove posts that are tagged with "wdc". I don't know how to do this in Ruby because [we are using kaminari for pagination](https://github.com/jfly/worldcubeassociation.org/blob/f661bc86fa8f0fcf850d18ea169ef68a239f21da/WcaOnRails/app/views/posts/index.html.erb#L3).